apps/jubmoji-quest/src/pages/recover/index.tsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useState } from "react";
+import { useRouter } from "next/navigation";
+import { detectIncognito } from "detectincognitojs";
+import { Jubmoji, succinctDeserializeJubmojiList } from "jubmoji-api";
+import { writeJubmojis } from "@/lib/localStorage";
+import { Card } from "@/components/cards/Card";
+import { CollectionCard } from "@/components/cards/CollectionCard";
+import { Button } from "@/components/ui/Button";
+import Link from "next/link";
+import { Icons } from "@/components/Icons";
+import { Modal } from "@/components/modals/Modal";
+import { useJubmojis } from "@/hooks/useJubmojis";
+
+export default function RecoverJubmojiPage() {
+  const router = useRouter();
+  const [isModalOpen, setIsModalOpen] = useState(true);
+  const { isLoading: isLoadingJubmojis, data: jubmojis = [] } = useJubmojis();
+  const [isLoadingSerialization, setIsLoadingSerialization] =
+    useState<boolean>(true);
+
+  // ensure users do not go to rest of app if they are in incognito
+  const alertIncognito = async () => {
+    const isIncognito = await detectIncognito();
+    if (isIncognito.isPrivate) {
+      alert(
+        "Please copy this link into a non-incognito tab in order to save your Jubmojis!"
+      );
+    }
+  };
+
+  useEffect(() => {
+    async function getCollectionFromUrl() {
+      const urlParams = new URLSearchParams(location.hash.slice(1));
+      const succinctSerializationURI = urlParams.get("collection");
+      if (!succinctSerializationURI) {
+        router.push("/");
+        return;
+      }
+      const succinctSerialization = decodeURIComponent(
+        succinctSerializationURI
+      );
+      const recoveredJubmojis = succinctDeserializeJubmojiList(
+        succinctSerialization
+      );
+
+      const unionJubmojis = new Set([...recoveredJubmojis, ...jubmojis]);
+      const unionJubmojisArray = Array.from(unionJubmojis);
+
+      // make sure we only keep the lowest nonce for each pubKeyIndex
```

Comment:
Any reason we go lowest instead of say, keeping the one in their current localStorage first?